analytic_engine/src/meta/meta_update.rs

```diff
@@ -457,7 +460,19 @@ pub struct MetaUpdateRequest {
 }
 
 impl MetaUpdateRequest {
-    pub fn new(location: WalLocation, meta_update: MetaUpdate) -> Self {
+    pub fn new(table_ident: TableIdentifier, meta_update: MetaUpdate) -> Self {
+        // Region id in manifest shouldn't change following by its moving from shards,
+        // so it should be mapped to `table_id`.
+        let versioned_region_id = VersionedRegionId {
+            version: table_ident.shard_info.cluster_version,
```

Comment:
Actually, I can't understand the necessity of the cluster version for `WalLocation`. Is it introduced to solve the problem caused by shard id reuse?

Reply:
The detail about this, you can see the pr introducing it #422